templates/src/addBook.js

```diff
@@ -1,40 +1,55 @@
 import React from 'react';
 import { connect } from 'react-redux';
-import { Link } from 'react-router-dom';
 import * as bookActions from './actions/index';
 
 class AddBook extends React.Component {
 
-  
-  submitBook(){
-    this.props.addBook(this.props.book);
+  submitBook() {
+    if (!this.isBookExist() && this.props.book.number) {
+      this.props.addBook(this.props.book);
+      this.props.clearBook();
+      this.props.history.push('/');
+    }
+  }
+
+  cancel() {
     this.props.clearBook();
+    this.props.history.push('/');
+  }
+
+  isBookExist() {
+    return this.props.books.some(book => book.number == this.props.book.number);
```

Comment:
It’s not the task of the component to check duplicates. Move the check outside the compost. It’s better to add duplicate field to props.